sktime/forecasting/trend.py

```diff
@@ -16,9 +16,103 @@
 from sktime.utils.datetime import _get_duration
 
 
-class PolynomialTrendForecaster(BaseForecaster):
+class TrendForecaster(BaseForecaster):
+    """Forecast time series data.
+
+    Default settings train a linear regression model.
+
+    Parameters
+    ----------
+    regressor : estimator object, optional (default = None)
+        Define the regression model type. If not set, will default to
+         sklearn.linear_model.LinearRegression
+
+    Example
+    ----------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.trend import TrendForecaster
+    >>> y = load_airline()
+    >>> forecaster = TrendForecaster()
+    >>> forecaster.fit(y)
+    TrendForecaster(...)
+    >>> y_pred = forecaster.predict(fh=[1,2,3])
     """
-    Forecast time series data with a polynomial trend.
+
+    _tags = {
+        "univariate-only": True,
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+    }
+
+    def __init__(self, regressor=None):
+        self.regressor = regressor
+        self.regressor_ = None
+        super(TrendForecaster, self).__init__()
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series with which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecast horizon with the steps ahead to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        # for default regressor, set fit_intercept=False
+        regressor = self.regressor or LinearRegression(fit_intercept=False)
+
+        # make pipeline with regressor
+        self.regressor_ = make_pipeline(regressor)
+
+        # transform data
+        n_timepoints = _get_duration(self._y.index, coerce_to_int=True) + 1
+        X = np.arange(n_timepoints).reshape(-1, 1)
```

Comment:
here, I'd take into account the case where time points are not equally spaced. The trend strategy still works in that case, you just have to be careful with reformatting.